### PR TITLE
Chore: Write test suite for SettingsHolder generated functions

### PR DESCRIPTION
**Pull request by brizental**
This is pre-work for my upcoming PR which actually completely refactors the Settings management.

The idea was to test the APIs as is, then change it all and check that the test are passing all the same.

Also, I removed an unused configuration for settings since I was at it.

### CODE REVIEW

File: tests/unit_tests/testsettingsholder.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "testsettingsholder.h"
+
+#define SETTING(type, toType, getter, setter, remover, has, key, defaultValue, \
```
Comment (Gela) [inline tests/unit_tests/testsettingsholder.cpp:7]: is the use of `\` necessary here? I looked at a few other files in the same directory and not seeing the same pattern (ex: testenv.cpp, testaddonindex.cpp, testchckedint.cpp)

File: tests/unit_tests/testsettingsholder.cpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                 expectedValue.join(","));                                     \
+        QVERIFY(settingsHolder.has());                                         \
+                                                                               \
+        settingsHolder.remover();                                              \
+        QCOMPARE(spy.count(), 2);                                              \
+        QVERIFY(!settingsHolder.has());                                        \
+        return;                                                                \
+      }                                                                        \
+      default: {                                                               \
+        Q_ASSERT(false);                                                       \
```
Comment (Gela) [inline tests/unit_tests/testsettingsholder.cpp:112]: Can more information be logged here?

File: tests/unit_tests/testsettingsholder.cpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        QCOMPARE(spy.count(), 2);                                              \
+        QVERIFY(!settingsHolder.has());                                        \
+        return;                                                                \
+      }                                                                        \
+      case QMetaType::Int:                                                     \
+      case QMetaType::LongLong: {                                              \
+        QVERIFY(!settingsHolder.has());                                        \
+        QCOMPARE(QVariant(settingsHolder.getter()).toInt(), v.toInt());        \
+                                                                               \
+        auto expectedValue = 42;                                               \
```
Comment (Gela) [inline tests/unit_tests/testsettingsholder.cpp:70]: Where does 42 come from?

File: tests/unit_tests/testsettingsholder.cpp (hunk 1 above)
Comment (brizental) [reply tests/unit_tests/testsettingsholder.cpp:7]: That is because this is macro.

File: tests/unit_tests/testsettingsholder.cpp (hunk 2 above)
Comment (brizental) [reply tests/unit_tests/testsettingsholder.cpp:112]: Given this is just a test case, do you think a comment would be enough?

File: tests/unit_tests/testsettingsholder.cpp (hunk 3 above)
Comment (brizental) [reply tests/unit_tests/testsettingsholder.cpp:70]: Just a random testing value.

**Review by strseb — APPROVED** [review]

**Review by Gela — APPROVED** [review]

**Pull request merged**